tests/send_rpc_reply_tests.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2025. ThingsBoard
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from time import sleep
+from unittest.mock import MagicMock, patch
+from tb_device_mqtt import TBDeviceMqttClient, RateLimit, TBPublishInfo, TBTimeoutException, TBQoSException
+from threading import RLock
+
+
+@patch('tb_device_mqtt.log')
+class TestTBDeviceMqttClientSendRpcReply(unittest.TestCase):
+    def setUp(self):
+        self.client = TBDeviceMqttClient(host="fake", port=0, username="", password="")
+
+    @patch.object(TBDeviceMqttClient, '_publish_data', autospec=True)
+    def test_send_rpc_reply_qos_invalid(self, mock_publish_data, mock_log):
+        result = self.client.send_rpc_reply("some_req_id", {"some": "response"}, quality_of_service=2)
+        self.assertIsNone(result)
+        mock_log.error.assert_called_with("Quality of service (qos) value must be 0 or 1")
```

Comment:
Using a logger message as a result of the test is not correct.
A test can be affected by simply changing the logger message but tested functionality will still be correct.